src/Services/NodeBalanceService.ts

```diff
@@ -1,8 +1,43 @@
 import {NodeBalance} from "../Models/NodeBalance";
+import sequelize from "../Services/Database";
+import {QueryTypes} from "sequelize";
+import {ServiceError} from "./ServiceError";
 
 export class NodeBalanceService {
 
     public async storeNodeBalance(balance: string, nodeId: number) {
         return await NodeBalance.create({balance, nodeId});
     }
+
+    public async fetchNodeBalance(nodeId: number) {
+        const balances = await sequelize.sequelize.query(
+            `(SELECT *
```

Comment:
1. Can you make use of Sequelize filters and functions? cc @MakMuftic 
2. I don't think you need UNION here. You can take all 24 hours results here and just use first and last in the array for the percentage difference.